test/e2e/e2e.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-sigs/kubebuilder/cmd/kubebuilder/util"
+	"github.com/kubernetes-sigs/kubebuilder/test/e2e/framework"
+	"github.com/kubernetes-sigs/kubebuilder/test/e2e/framework/ginkgowrapper"
+	e2einternal "github.com/kubernetes-sigs/kubebuilder/test/internal/e2e"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// RunE2ETests checks configuration parameters (specified through flags) and then runs
+// E2E tests using the Ginkgo runner.
+func RunE2ETests(t *testing.T) {
+	RegisterFailHandler(ginkgowrapper.Fail)
+	glog.Infof("Starting kubebuilder suite")
+	RunSpecs(t, "Kubebuilder e2e suite")
+}
+
+var _ = Describe("main workflow", func() {
+	It("should perform main kubebuilder workflow successfully", func() {
+		testSuffix := framework.RandomSuffix()
+		c := initConfig(testSuffix)
+		kubebuilderTest := e2einternal.NewKubebuilderTest(c.workDir, framework.TestContext.BinariesDir)
+
+		prepare(c.workDir)
+		defer cleanup(kubebuilderTest, c.workDir, c.controllerImageName)
+
+		var controllerPodName string
+		var e error
+
+		By("init project")
+		initOptions := []string{"--domain", c.domain}
+		err := kubebuilderTest.Init(initOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating resource definition")
+		resourceOptions := []string{"--group", c.group, "--version", c.version, "--kind", c.kind}
+		err = kubebuilderTest.CreateResource(resourceOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating core-type resource controller")
+		controllerOptions := []string{"--group", "apps", "--version", "v1beta2", "--kind", "Deployment", "--core-type"}
+		err = kubebuilderTest.CreateController(controllerOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("building image")
+		// The scaffold test cases generated for core types cannot work without manually modification.
+		// See https://github.com/kubernetes-sigs/kubebuilder/pull/193 for more details
+		// Skip the test part in build process as we don't care about it here.
```

Comment:
It might be better to simply delete the `go test` for the scaffolded controller that doesn't pass.

Reply:
done